File: uninstall_hook.py
#!/usr/bin/env python3
"""Continuum — Claude Code auto-context uninstaller.

Reverses install_hook.py: removes the UserPromptSubmit and SessionEnd hook
scripts, their settings.json registrations, and the local token/workspace state.
Safe to re-run (idempotent), safe to run even if never installed, and cleans up
installs from older bash-only versions too.

  curl -fsSL https://continuum-mcp.sshogunn.org/uninstall-hook.sh | bash
"""

import json
import shutil
import sys
from pathlib import Path

HOME = Path.home()
STATE_DIR = HOME / ".continuum"
HOOKS_DIR = HOME / ".claude" / "hooks"
SETTINGS_PATH = HOME / ".claude" / "settings.json"

STEMS = ("continuum-context-inject", "continuum-session-capture")
EVENTS = ("UserPromptSubmit", "SessionEnd")


def remove_files():
    for stem in STEMS:
        for path in (HOOKS_DIR / (stem + ".py"), HOOKS_DIR / stem):
            try:
                path.unlink()
            except OSError:
                pass
    shutil.rmtree(STATE_DIR, ignore_errors=True)


def clean_settings():
    try:
        with open(SETTINGS_PATH, encoding="utf-8") as f:
            settings = json.load(f)
    except FileNotFoundError:
        return None
    except ValueError:
        print(
            "error: %s has invalid JSON — skipping settings cleanup" % SETTINGS_PATH,
            file=sys.stderr,
        )
        return None

    hooks = settings.get("hooks", {})
    removed = False

    for event in EVENTS:
        groups = hooks.get(event)
        if not groups:
            continue
        kept = []
        for group in groups:
            original = group.get("hooks", [])
            remaining = [
                entry
                for entry in original
                if not any(stem in str(entry.get("command", "")) for stem in STEMS)
            ]
            if len(remaining) != len(original):
                removed = True
            if remaining or not original:
                group["hooks"] = remaining
                kept.append(group)
        if kept:
            hooks[event] = kept
        else:
            hooks.pop(event, None)

    with open(SETTINGS_PATH, "w", encoding="utf-8") as f:
        json.dump(settings, f, indent=2)
        f.write("\n")
    return removed


def main():
    remove_files()
    removed = clean_settings()
    print("")
    if removed is None:
        print("no settings.json to clean up")
    else:
        print("settings.json entries removed" if removed else "no matching settings.json entries found")
    print(
        "Continuum hooks uninstalled: context injection, session capture, local token, "
        "and workspace state removed."
    )
    print("Run /hooks in Claude Code (or restart it) to pick up the change.")


if __name__ == "__main__":
    main()

File: hooks/uninstall_hook.js
#!/usr/bin/env node
"use strict";

// Continuum — Claude Code auto-context uninstaller.
//
// Reverses install_hook.js: removes the UserPromptSubmit and SessionEnd hook
// scripts (including any leftover files from the older Python-based install),
// their settings.json registrations, and the local token/workspace state.
// Safe to re-run (idempotent), safe to run even if never installed.
//
//   Linux/macOS   curl -fsSL https://continuum-mcp.sshogunn.org/uninstall_hook.js | node
//   Windows       irm https://continuum-mcp.sshogunn.org/uninstall_hook.js | node -

const fs = require("fs");
const os = require("os");
const path = require("path");

const HOME = os.homedir();
const STATE_DIR = path.join(HOME, ".continuum");
const HOOKS_DIR = path.join(HOME, ".claude", "hooks");
const SETTINGS_PATH = path.join(HOME, ".claude", "settings.json");

const STEMS = ["continuum-context-inject", "continuum-session-capture", "continuum-self-update"];
const EVENTS = ["UserPromptSubmit", "SessionEnd"];

function removeFiles() {
  for (const stem of STEMS) {
    for (const ext of [".js", ".py", ""]) {
      try {
        fs.unlinkSync(path.join(HOOKS_DIR, stem + ext));
      } catch {
        // wasn't there — fine
      }
    }
  }
  fs.rmSync(STATE_DIR, { recursive: true, force: true });
}

function cleanSettings() {
  let settings;
  try {
    settings = JSON.parse(fs.readFileSync(SETTINGS_PATH, "utf-8"));
  } catch (exc) {
    if (exc.code === "ENOENT") return null;
    process.stderr.write(`error: ${SETTINGS_PATH} has invalid JSON — skipping settings cleanup\n`);
    return null;
  }

  const hooks = settings.hooks || {};
  let removed = false;

  for (const event of EVENTS) {
    const groups = hooks[event];
    if (!groups) continue;
    const kept = [];
    for (const group of groups) {
      const original = group.hooks || [];
      const remaining = original.filter(
        (entry) => !STEMS.some((stem) => String(entry.command || "").includes(stem))
      );
      if (remaining.length !== original.length) removed = true;
      if (remaining.length || !original.length) {
        group.hooks = remaining;
        kept.push(group);
      }
    }
    if (kept.length) hooks[event] = kept;
    else delete hooks[event];
  }

  fs.writeFileSync(SETTINGS_PATH, JSON.stringify(settings, null, 2) + "\n");
  return removed;
}

function main() {
  removeFiles();
  cleanSettings();
  console.log("");
  console.log("Continuum has been removed, along with your saved token on this machine.");
  console.log("Your memories are untouched — sign in to the dashboard anytime.");
  console.log("Restart Claude Code to finish.");
}

main();
